src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const add = (val) => {
+    if (val !== undefined) {
+      sum = val;
+
+      return (arg) => {
+        if (arg !== undefined) {
+          return add(sum + arg);
```

Comment:
Instead of returning the result of add(sum + arg), you should just add arg to the sum and return the add function.